internal/mode/static/state/change_processor_test.go

```diff
@@ -1523,6 +1524,60 @@ var _ = Describe("ChangeProcessor", func() {
 				})
 			})
 		})
+
+		Describe("NginxProxy resource changes", Ordered, func() {
```

Comment:
Could we also include NginxProxy in tests for "Multiple Gateway API resource changes" ? Those check for relatedness